crates/circuit/src/classical/expr/expr.rs

```diff
@@ -168,6 +168,82 @@ impl Expr {
         }
         visitor(self.as_mut())
     }
+
+    /// Do these two expressions have exactly the same tree structure?
```

Comment:
It's the first time I've seen docstrings being written this way 😄, though I'm not quite sure if it is proper since we've been formatting docstrings inconsistently when it comes to Rust.

Reply:
It's mostly just ported from the docstring we had in the Python code (that one is below, verbatim). I just tweaked it to remove the var key stuff since in this version of the method that is not specifiable.